src/modules/fancyzones/lib/FancyZonesData.cpp

```diff
@@ -95,11 +95,24 @@ std::optional<FancyZonesDataTypes::CustomZoneSetData> FancyZonesData::FindCustom
 
 void FancyZonesData::AddDevice(const std::wstring& deviceId)
 {
+    using namespace FancyZonesDataTypes;
+
     std::scoped_lock lock{ dataLock };
     if (!deviceInfoMap.contains(deviceId))
     {
         // Creates default entry in map when ZoneWindow is created
-        deviceInfoMap[deviceId] = FancyZonesDataTypes::DeviceInfoData{ FancyZonesDataTypes::ZoneSetData{ NonLocalizable::NullStr, FancyZonesDataTypes::ZoneSetLayoutType::Blank } };
+        GUID guid;
+        auto result{ CoCreateGuid(&guid) };
+        wil::unique_cotaskmem_string guidString;
+        if (result == S_OK && SUCCEEDED(StringFromCLSID(guid, &guidString)))
+        {
+            DeviceInfoData defaultDeviceInfoData{ ZoneSetData{ guidString.get(), ZoneSetLayoutType::PriorityGrid }, true, 16, 3 };
+            deviceInfoMap[deviceId] = std::move(defaultDeviceInfoData);
+        }
+        else
+        {
+            deviceInfoMap[deviceId] = DeviceInfoData{ ZoneSetData{ NonLocalizable::NullStr, ZoneSetLayoutType::Blank } };
```

Comment:
Should we add anything into `deviceInfoMap` in this case? I remember that `ZoneSetLayoutType::Blank` has been added for some purpose, but I can't recall why exactly.  
Everywhere there are checks if zone set UUID is valid and `ZoneSetLayoutType` is not `Blank`. It won't be saved in the settings file and just takes a place in the memory.

Reply:
I'll double check whether `Blank` type and checks are still needed at all

Reply:
Keep in mind that we will add functionality to the editor to set no zoneset on a desktop.

Reply:
🔼 That's why it is still needed :)